lib/cppcheck.cpp

```diff
@@ -1092,8 +1087,13 @@ unsigned int CppCheck::checkInternal(const FileWithDetails& file, const std::str
 
             // Check only a few configurations (default 12), after that bail out, unless --force
             // was used.
-            if (!mSettings.force && ++checkCount > mSettings.maxConfigs)
+            if (!mSettings.force && ++checkCount > mSettings.maxConfigs) {
+                // if maxConfigs has default value then report information message that configurations are skipped
+                if (mSettings.maxConfigs == 12 && mSettings.severity.isEnabled(Severity::information))
+                    tooManyConfigsError(Path::toNativeSeparators(file.spath()), configurations.size());
+
```

Comment:
This feels very strange.

Also the check for the default value should not be hard-coded - use a default `Settings` object to check against.